src/mgr/ActivePyModules.cc

```diff
@@ -790,11 +791,29 @@ PyObject *ActivePyModules::get_osdmap()
 void ActivePyModules::set_health_checks(const std::string& module_name,
 				  health_check_map_t&& checks)
 {
-  Mutex::Locker l(lock);
+  bool changed = false;
+
+  lock.Lock();
   auto p = modules.find(module_name);
   if (p != modules.end()) {
-    p->second->set_health_checks(std::move(checks));
+    changed = p->second->set_health_checks(std::move(checks));
   }
+  lock.Unlock();
+
+  // immediately schedule a report to be sent to the monitors with the new
+  // health checks that have changed. This is done asynchronusly to avoid
+  // blocking python land. ActivePyModules::lock needs to be dropped to make
+  // lockdep happy:
+  //
+  //   send_report callers: DaemonServer::lock -> PyModuleRegistery::lock
```

Comment:
Yeah, this is an artifact of the MonMgrReport being sent from DaemonServer -- it doesn't really belong there, as it's not a server-ish thing.  We should follow up at some point with a refactor to send all the reports/beacons from Mgr or from a new Beacon class (similar to what MDS does), so that this kind of bidirectional stuff doesn't come up.